lib/server/swarm.js

```diff
@@ -98,15 +108,23 @@ Swarm.prototype._onAnnounceUpdate = function (params, peer) {
     this.complete += 1
     this.incomplete -= 1
     peer.complete = true
+    var id = params.type === 'ws' ? params.peer_id : params.addr
+    this.peers.set(id, peer)
   }
 }
 
 Swarm.prototype._getPeers = function (numwant, ownPeerId, isWebRTC) {
   var peers = []
-  var ite = randomIterate(Object.keys(this.peers))
+  var ite = randomIterate(Object.keys(this.peers.cache))
   var peerId
   while ((peerId = ite()) && peers.length < numwant) {
-    var peer = this.peers[peerId]
+    // Check manually if the peer is active
+    if (peers.maxAge && (Date.now() - peers.cache[peerId].modified) > peers.maxAge) {
```

Comment:
Maybe this deserves a PR to LRU too, I am trying to access an element without bringing it up in the chain but with evicting it if its too old.